### PR TITLE
fix Program Service not reachable as a docker container

### PR DESCRIPTION
**Pull request by headsoup**
**Description of changes**

Adds a few goodies to facilitate implementation and troubleshooting.
These were useful trying to figure out a cryptic set of errors when a feature was deployed without its corresponding environment variables. It also covers a few bugs found along the way (mostly TS migration tasks).

**Type of Change**

- [ ] Bug
- [x] New Feature

### CODE REVIEW

File: .eslintrc.js
Comment (headsoup) [inline .eslintrc.js:1]: ESLint is your friend. This will help with the remaining `.js` files in the same manner VSCode helps in TS files, as well as suggest extra improvements to the code, e.g. order of inputs.

File: .eslintrc.js
Comment (headsoup) [inline .eslintrc.js:52]: TODO: The absolute internal imports should have a prefix, to make them more declarative

File: Dockerfile
Comment (headsoup) [inline Dockerfile:1]: changes in this file are prep-work towards updating node version.
tried to make it node v18, but had a few hurdles, so I prioritised getting this app back online first.

File: Dockerfile
Comment (headsoup) [inline Dockerfile:16]: not needed since now we can use ts-node in prod. saves jenkins time too.

File: Jenkinsfile.groovy
Comment (headsoup) [inline Jenkinsfile.groovy:96]:
```suggestion
```

to be removed as soon as the PR is approved. I left it in case there's feedback.

File: README.md
Comment (headsoup) [inline README.md:15]: can we not suggest people do `npm i` anymore? makes life harder for maintainers.

File: package.json
Comment (headsoup) [inline package.json:12]: yep. ts-node in prod. rejoice!

File: package.json
Comment (headsoup) [inline package.json:14]: these lines make absolute internal modules possible

File: package.json
Comment (headsoup) [inline package.json:108]: to cover testing with the absolute internal imports

File: src/routes/file-centric-tsv/utils.test.ts
Comment (headsoup) [inline src/routes/file-centric-tsv/utils.test.ts:1]: changes in this file were to ensure this test stops failing randomly. rather frustrating

File: src/routes/file-centric-tsv/utils.test.ts
Comment (headsoup) [inline src/routes/file-centric-tsv/utils.test.ts:111]:
```suggestion
```

oops

File: src/schemas/Program/index.js
Comment (headsoup) [inline src/schemas/Program/index.js:326]: these changes were just to make the intent easier to reason with

File: src/utils/logger.ts
Comment (headsoup) [inline src/utils/logger.ts:1]: WIP. the pod wasn't showing some logs, and it made troubleshooting rather hard.

File: tsconfig.json
Comment (headsoup) [inline tsconfig.json:1]: change spaces to tabs, and some enhancements to make dev life easier.

**Review by headsoup — COMMENTED** [review]
Some comments to help explain intent of changes, and facilitate review.

**Review by samrichca — APPROVED** [review]
thank you for fixing this, the changes look good to me

**Pull request merged**
